api/middleware/auth0Middleware.js

```diff
@@ -0,0 +1,64 @@
+const { expressjwt: jwt } = require('express-jwt');
+const jwksRsa = require('jwks-rsa');
+const createError = require('http-errors');
+const config = require('../../config/auth0');
+// const { findOrCreateProfile } = require('../profile/profileModel');
+
+// const makeProfileObj = (claims) => {
+//   return {
+//     id: claims.sub.slice(6),
+//     email: claims.email,
+//     name: claims.name,
+//   };
+// };
+
+// writing the config for expressJwt method to be invoked with product environment variables
+const verifyJwt = jwt({
+  secret: jwksRsa.expressJwtSecret({
+    cache: true,
+    rateLimit: true,
+    jwksRequestsPerMinute: 5,
+    jwksUri: config.jwksUri,
+  }),
+  audience: config.audience,
+  issuer: config.issuer,
+  algorithms: ['RS256'],
+  requestProperty: 'auth0User',
+}).unless({ path: ['/'] });
```

Comment:
Now the user can visit any public endpoints that will be in the array. Great job there!